modules/misc/nix.nix

```diff
@@ -221,6 +246,15 @@ in {
       '';
     }];
 
+    home = let NIX_PATH = concatStringsSep ":" cfg.nixPath;
+    in {
+      sessionVariables = mkIf (!cfg.keepOldNixPath) { inherit NIX_PATH; };
+
+      sessionVariablesExtra = mkIf (cfg.keepOldNixPath && cfg.nixPath != [ ]) ''
+        export NIX_PATH="${NIX_PATH}''${NIX_PATH:+:$NIX_PATH}"
```

Comment:
I think the semantics of [`home.sessionVariables`](https://nix-community.github.io/home-manager/options.html#opt-home.sessionVariables) means that you can use

```nix
home.sessionVariables.NIX_PATH = "${NIX_PATH}\${NIX_PATH:+:$NIX_PATH}";
```

Since it says:

> The values may refer to other environment variables using POSIX.2 style variable references

Instead of using `home.sessions.sessionVariablesExtra`

Reply:
Also, double-quoted strings use `\` to escape the `${text}` syntax, not `''` like indented strings.